src/DynamoCore/Interfaces/IPathResolver.cs

```diff
@@ -70,9 +70,15 @@ public interface IPathManager
         string CommonDataDirectory { get; }
 
         /// <summary>
-        /// The local directory that contains custom nodes created by the user.
+        /// Directories from where custom nodes are to be loaded. The implementor
+        /// of this interface method should always guarantee that a non-empty 
+        /// list is returned, and that the first entry represents the default 
+        /// custom node directory. Custom nodes created are stored in the
+        /// default directory, which is specific to the current user. Changes to
+        /// custom nodes may or may not be saved to their current location depeding
+        /// on write access.
         /// </summary>
-        string UserDefinitions { get; }
+        IEnumerable<string> UserDefinitionDirectories { get; }
```

Comment:
Please check with Ritesh to see if this is needed. We will not have separated folders for `packages` and `definitions`. All under these custom folders are just loose dyf and package folders like `Mantis`